atlasdb-commons/src/main/java/com/palantir/async/initializer/AsyncInitializer.java

```diff
@@ -43,26 +43,50 @@
     private final AtomicBoolean isInitializing = new AtomicBoolean(false);
     private volatile boolean initialized = false;
     private volatile boolean canceledInitialization = false;
+    private int numberOfInitializationAttempts = 1;
+    private Long initializationStartTime;
 
     public final void initialize(boolean initializeAsync) {
         assertNeverCalledInitialize();
 
+        initializationStartTime = System.currentTimeMillis();
+
         if (!initializeAsync) {
             tryInitializeInternal();
             return;
         }
 
+        tryInitializationLoop();
+    }
+
+    private void tryInitializationLoop() {
         try {
             tryInitializeInternal();
-            log.info("Initialized {} synchronously.", SafeArg.of("className", getInitializingClassName()));
+            log.info("Initialized {} on the {} attempt in {} milliseconds",
```

Comment:
To avoid having to put something like `st/nd/rd/th` or `attempt(s)`, you could put `on attempt {}` (ditto below).